### PR TITLE
Add test for verifying if route selection takes both transfer amount as well as fees into account

### PR DESCRIPTION
**Pull request by lullis**
Fixes: #4253 

When introducing source routing, one issue was found regarding checking if the channel had enough funds to cover both the transfer as well as the mediator fees. This PR adds unit tests to ensure we have no regression.

### CODE REVIEW

File: raiden/tests/unit/transfer/test_source_routing.py [hunk 1]
```diff
@@ -152,6 +153,69 @@ def test_resolve_routes(netting_channel_state, chain_state, token_network_state)
     assert route_states[0].forward_channel_id == channel_id, msg
 
 
+def test_initiator_accounts_for_fees_when_selecting_routes():
+    def make_mediated_transfer_state_change(
+        transfer_amount, allocated_fee_amount, channel_capacity
```
Comment (palango) [inline raiden/tests/unit/transfer/test_source_routing.py]: Would be nice to have type annotations, including the return type.

File: raiden/tests/unit/transfer/test_source_routing.py [hunk 2]
```diff
@@ -152,6 +153,69 @@ def test_resolve_routes(netting_channel_state, chain_state, token_network_state)
     assert route_states[0].forward_channel_id == channel_id, msg
 
 
+def test_initiator_accounts_for_fees_when_selecting_routes():
```
Comment (palango) [inline raiden/tests/unit/transfer/test_source_routing.py]: Small docstring with the description in the PR would be nice.

File: raiden/tests/unit/transfer/test_source_routing.py [hunk 3]
```diff
@@ -152,6 +153,69 @@ def test_resolve_routes(netting_channel_state, chain_state, token_network_state)
...
+    underfunded_channel = make_mediated_transfer_state_change(
+        transfer_amount=10, allocated_fee_amount=0, channel_capacity=9
+    )
+    assert search_for_item(underfunded_channel.events, EventPaymentSentFailed, {}) is not None
+
+    # This channel has enough balance to cover the transfer
+    funded_channel = make_mediated_transfer_state_change(
+        transfer_amount=10, allocated_fee_amount=2, channel_capacity=12
+    )
+    assert search_for_item(funded_channel.events, EventPaymentSentFailed, {}) is None
```
Comment (palango) [inline raiden/tests/unit/transfer/test_source_routing.py]: Can you also add a positive assert here? IOW, check that some event happened?

File: raiden/tests/unit/transfer/test_source_routing.py (hunk 1 above)
Comment (lullis) [reply raiden/tests/unit/transfer/test_source_routing.py]: Done

File: raiden/tests/unit/transfer/test_source_routing.py (hunk 3 above)
Comment (lullis) [reply raiden/tests/unit/transfer/test_source_routing.py]: Added check for SendLockedTransfer event

**Pull request merged**